File: 2010_2011_sujet_2_simulation_de_lois/discrete1.m
function res=discrete1(n)
% Remarque : 5/6 = 1/2+ 1/3
X=rand(1,n);
res=(X<1/2) + 2*(X>1/2 & X<5/6) + 3*(X>5/6);
% la ligne precedente renvoit 1 si X<1/2, 2 si 1/2<X<(1/2+1/3), 3 si
% X>1/2+1/3
end

File: 2011_2012_sujet_2_simulation_de_lois/discrete1.py
import numpy as np


def discrete1(n):
    X = np.random.rand(n)
    return (X < 1/2).astype(int) + 2*((X > 1/2) & (X < 5/6)).astype(int) + 3*(X > 5/6).astype(int)
